### ISSUE TITLE
How to manage accessibility focus with modals

### ISSUE THREAD

**Issue by vodemn** [flutter/flutter#107069]
## Steps to Reproduce

1. Turn on VoiceOver
2. Navigate to 'Open dialog' button
3. Close dialog using 'Close dialog' button

**Expected results:** After closing the dialog accessibility focus should be back on the button, that opened this dialog.

**Actual results:** Accessibility focus is on the whole page.

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() {
  RendererBinding.instance?.setSemanticsEnabled(true);
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      debugShowCheckedModeBanner: false,
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  final String title;

  const MyHomePage({
    Key? key,
    required this.title,
  }) : super(key: key);

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  final GlobalKey _buttonKey = GlobalKey();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Semantics(
          key: _buttonKey,
          label: 'Open dialog',
          button: true,
          excludeSemantics: true,
          onTap: _openDialog,
          child: ElevatedButton(
            onPressed: _openDialog,
            child: const Text('Open dialog'),
          ),
        ),
      ),
    );
  }

  void _openDialog() {
    showDialog(
      context: context,
      builder: (context) => SimpleDialog(
        children: [
          Semantics(
            label: 'Close dialog',
            button: true,
            excludeSemantics: true,
            onTap: Navigator.of(context).pop,
            child: ElevatedButton(
              onPressed: Navigator.of(context).pop,
              child: const Text('Close'),
            ),
          ),
        ],
      ),
    );
  }
}

```

</details>
